lib/improver/tests/test_generate_ancillaries_topographicmasks.py

```diff
@@ -0,0 +1,164 @@
+# All rights reserved.
+#
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are met:
+#
+# * Redistributions of source code must retain the above copyright notice, this
+#   list of conditions and the following disclaimer.
+#
+# * Redistributions in binary form must reproduce the above copyright notice,
+#   this list of conditions and the following disclaimer in the documentation
+#   and/or other materials provided with the distribution.
+#
+# * Neither the name of the copyright holder nor the names of its
+#   contributors may be used to endorse or promote products derived from
+#   this software without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Unit tests for the generate_ancillary.GenerateOrogBandAncils plugin."""
+
+
+import unittest
+from iris.cube import Cube
+from iris.tests import IrisTest
+from iris.exceptions import CoordinateNotFoundError
+import numpy as np
+
+from improver.generate_ancillaries.generate_ancillary import (
+    GenerateOrographyBandAncils as GenOrogMasks)
+
+
+class Test_gen_orography_masks(IrisTest):
+    """
+    Test the gen_orography_masks method orography band mask
+    ancillary generation plugin.
+    """
+    def setUp(self):
+        """setting up test input and output data sets"""
+        landmask_data = np.array([[0.25, 0., 0.],
+                                  [0.75, 0.25, 0.],
+                                  [1., 1., 0.75]])
+        self.landmask = Cube(landmask_data, long_name='test land')
+        orog_data = np.array([[10., 0., 0.],
+                              [20., 100., 15.],
+                              [-10., 100., 40.]])
+        self.orography = Cube(orog_data, long_name='test orog')
+
+    def test_nonsensekey(self):
+        """test the correct exception is raised for unknown keys"""
+        exception_dict = {'nonsense': [[0, 10]]}
+        with self.assertRaisesRegexp(KeyError, 'Unknown threshold_dict key'):
+            GenOrogMasks().gen_orography_masks(
+                self.orography, self.landmask, "nonsense",
+                exception_dict["nonsense"])
+
+
+class Test_process(IrisTest):
```

Comment:
I think that most of the tests here on the `process` method are actually testing functionality within the `gen_orography_masks` method. Could you redistribute these tests?